src/contracts/permissions/PermissionControllerStorage.sol

```diff
@@ -0,0 +1,30 @@
+// SPDX-License-Identifier: BUSL-1.1
+pragma solidity ^0.8.27;
+
+import "@openzeppelin/contracts/utils/structs/EnumerableSet.sol";
+
+import "../interfaces/IPermissionController.sol";
+
+abstract contract PermissionControllerStorage is IPermissionController {
+    using EnumerableSet for EnumerableSet.Bytes32Set;
+    using EnumerableSet for EnumerableSet.AddressSet;
+
+    struct AccountPermissions {
```

Comment:
I quite like this storage design

Reply:
lets go the rest of the way with this pattern and used salted storage location. Then we don't have to worry about storage gaps

Reply:
Prefer not using salted location for consistency with storage layouts across core

Reply:
https://github.com/Layr-Labs/eigenlayer-contracts/pull/870/commits/eac8bbe1116ca3cef86eaada1f9d6d666aff49f7